src/com/android/calendar/CalendarToolbarHandler.java

```diff
@@ -141,7 +141,14 @@ private String buildDayOfWeek() {
             dayOfWeek = DateUtils.formatDateRange(mContext, mFormatter, mMilliTime, mMilliTime,
                     DateUtils.FORMAT_SHOW_WEEKDAY, mTimeZone).toString();
         }
-        return dayOfWeek.toUpperCase();
```

Comment:
(I would propose to stay with English, so others can follow).

For me it shows "Heute, Sonntag" tried Android 4.4.4 and 5.1